polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -282,14 +275,27 @@ void MemorySelector::analyze(LoopLikeOpInterface loop, AccessKind accessKind) {
           bool strideIsLargeEnough = stride->sgt(8) || stride->slt(-8);
           useSharedMemory = hasTemporalReuse(access, threadVars) &&
                             (stride->isZero() || strideIsLargeEnough);
-        }
-
-        accessToMemSpace[&access] =
+          // FIXME: getConstIntegerValue doesn't return zero for
```

Comment:
We have this "recovery code" in another place (isEqual?). Makes me think we could place it in the `getConstIntegerValue` so consumer do not have to all add this code.